sdk/resourcemanager/Azure.ResourceManager/src/Resources/Generated/ResourceLink.cs

```diff
@@ -56,23 +56,20 @@ internal ResourceLink(ArmResource options, ResourceLinkData data) : base(options
         /// <param name="id"> The identifier of the resource that is the target of operations. </param>
         internal ResourceLink(ArmResource options, ResourceIdentifier id) : base(options, id)
         {
-            _clientDiagnostics = new ClientDiagnostics(ClientOptions);
-            _resourceLinksRestClient = new ResourceLinksRestOperations(_clientDiagnostics, Pipeline, ClientOptions, BaseUri);
+            _clientDiagnostics = new ClientDiagnostics("Azure.ResourceManager.Resources", ResourceType.Namespace, DiagnosticOptions);
+            _resourceLinksRestClient = new ResourceLinksRestOperations(_clientDiagnostics, Pipeline, DiagnosticOptions.ApplicationId, BaseUri);
 #if DEBUG
 			ValidateResourceId(Id);
 #endif
         }
 
         /// <summary> Initializes a new instance of the <see cref="ResourceLink"/> class. </summary>
-        /// <param name="clientOptions"> The client options to build client context. </param>
-        /// <param name="credential"> The credential to build client context. </param>
-        /// <param name="uri"> The uri to build client context. </param>
-        /// <param name="pipeline"> The pipeline to build client context. </param>
+        /// <param name="client"> The client options to build client context. </param>
         /// <param name="id"> The identifier of the resource that is the target of operations. </param>
-        internal ResourceLink(ArmClientOptions clientOptions, TokenCredential credential, Uri uri, HttpPipeline pipeline, ResourceIdentifier id) : base(clientOptions, credential, uri, pipeline, id)
+        internal ResourceLink(ArmClient client, ResourceIdentifier id) : base(client, id)
```

Comment:
The implementation of these ctors is pretty almost the same for all of them. Do you think it can be unified in some way?

Reply:
yeah I did that to this one as an example, but since this is all autogen didn't want to manually update each one 
https://github.com/Azure/azure-sdk-for-net/pull/26452/files#diff-511c3c1222911b5c9e0d3f3284200afbab40cb89f6cbc669e5727b01626e8d8c